Cython/Build/IpythonMagic.py

```diff
@@ -106,6 +108,43 @@ def encode_fs(name):
         return name
 
 
+def get_encoding_candidates():
+    candidates = [sys.getdefaultencoding()]
+    try:
+        candidates.extend([sys.stdout.encoding, sys.stdin.encoding])
+    except AttributeError:
+        pass
+    try:
+        candidates.extend([sys.__stdout__.encoding, sys.__stdin__.encoding])
+    except AttributeError:
+        pass
+    unique_candidates = OrderedDict.fromkeys(candidates)
+    # encoding might be None (i.e. somebody redirects stdout):
+    unique_candidates.pop(None, None)
+    return list(unique_candidates)
+
+
+def prepare_captured(captured):
+    captured_bytes = captured and captured().strip()
+    if not captured_bytes:
+        return None
+    for encoding in get_encoding_candidates():
+        try:
+            return captured_bytes.decode(encoding)
+        except UnicodeDecodeError:
+            pass
+    # last resort: print at least the readable ascii parts correctly.
+    return captured_bytes.decode('latin-1')
+
+
+def print_captured(captured, output, header_line):
```

Comment:
If it's optional, make it optional.
```suggestion
def print_captured(captured, output, header_line=None):
```

Reply:
Done: https://github.com/cython/cython/pull/3872/commits/d9c07d6f315ac1c054381a1637907c97d660ccd5